tools/rosbag/scripts/makerule.py

```diff
@@ -42,6 +42,12 @@
 import genpy.message
 import genpy.dynamic
 
+# raw_input was renamed to input in Python 3.x
+try:
+    input = raw_input  # Python 2.x: Simulate 3.x behavior.
+except NameError:
+    pass  # Python 3.x: There is no raw_input, use input directly.
```

Comment:
Do we need to make this backwards compatible?

As I understand, since ROS Noetic we're only expected to support Python 3 (see [REP 3](https://www.ros.org/reps/rep-0003.html#noetic-ninjemys-may-2020-may-2025).

cc/ @sloretz

Reply:
Correct, on the noetic branch there's no need for the backwards compatibility. Some people use Python 3 with older ROS distros (like arch melodic users), so mabye @bst added it in hopes for a backport to melodic?

Reply:
Ah, i wasn't aware of that. thanks for pointing that out 👍

Reply:
I would remove the workaround here, and if there is desire to backport to Melodic, we can consider the workaround in a backport PR.

Reply:
agree. we would not want to keep something will never be used.

Reply:
Makes sense to me. I just removed the python2 compatibility workaround I had added before.